bolt-cli/src/commands/register.rs

```diff
@@ -0,0 +1,127 @@
+use std::collections::HashMap;
+
+use alloy::{
+    network::EthereumWallet,
+    primitives::{address, Address},
+    providers::ProviderBuilder,
+    signers::local::PrivateKeySigner,
+    sol,
+};
+use ethereum_consensus::crypto::PublicKey as BlsPublicKey;
+use eyre::Context;
+
+use crate::{
+    cli::{Chain, RegisterCommand},
+    common::hash::compress_bls_pubkey,
+};
+
+#[derive(Debug, PartialEq, Eq, Hash)]
+enum BoltContract {
+    Validators,
+}
+
+impl RegisterCommand {
+    /// Run the `delegate` command.
+    pub async fn run(self) -> eyre::Result<()> {
+        let bolt_validators_address = bolt_validators_address(self.chain);
+
+        let pubkeys_file = std::fs::File::open(&self.pubkeys_path)?;
+        let keys: Vec<BlsPublicKey> = serde_json::from_reader(pubkeys_file)?;
+        let pubkey_hashes: Vec<_> = keys.iter().map(compress_bls_pubkey).collect();
+
+        let wallet: PrivateKeySigner =
+            self.admin_private_key.parse().wrap_err("invalid private key")?;
+        let transaction_signer = EthereumWallet::from(wallet);
+        let provider = ProviderBuilder::new()
+            .with_recommended_fillers()
+            .wallet(transaction_signer)
+            .on_http(self.rpc_url.clone());
+
+        let bolt_validators = BoltValidatorsContract::new(bolt_validators_address, provider);
+
+        let call = bolt_validators.batchRegisterValidatorsUnsafe(
+            pubkey_hashes,
+            self.max_committed_gas_limit,
+            self.authorized_operator,
+        );
+
+        let result = call.send().await?.watch().await?;
+
+        println!("transaction hash: {:?}", result);
+
+        Ok(())
+    }
+}
+
+// PERF: this should be done at compile time
```

Comment:
we could use phf for this https://docs.rs/phf/latest/phf/ but I think it's way overkill and I would opt for not adding extra dependencies. what do you think?

Reply:
Opting for a hashmap + struct approach.